### ISSUE TITLE
Can't get Android modules with Engine.get_singleton() anymore

### ISSUE THREAD

**Issue by samuelpedrajas** [godotengine/godot#22859]
**Godot version:**

3.1 Alpha commit 0063ba92075094b7cc9d36f4551080b89e0c0e50

**OS/device including version:**

Android 7 and 9/ Xiaomi MI5 and Galaxy Tab A8 (2016)

**Issue description:**

I used to call `Engine.has_singleton("SomeModule")` and `Engine.get_singleton("SomeModule")` to use Android modules in my game, but a few days ago I pulled the master branch and these two functions stopped behaving as before. I always get `False` when calling `Engine.has_singleton("SomeModule")` and `[Object:null]` when calling `Engine.get_singleton("SomeModule")`.

I don't know too much about the internals of the engine, but after debugging a bit I found that these two lines were removed from `platform/android/java_glue.cpp` at the end of the `_initialize_java_modules` function:

```
jobject obj = env->CallStaticObjectMethod(singletonClass, initialize, _godot_instance);
jobject gob = env->NewGlobalRef(obj);
```

For some reason if I get back the first line (`env->CallStaticObjectMethod(singletonClass, initialize, _godot_instance);`) it gets fixed, but I'm not sure if this is the right solution.

**Steps to reproduce:**
I used the example of this module https://github.com/Shin-NiL/Godot-Share/ as a minimal reproduction project. I rewrote the code in `_on_share_btn_pressed` like this:

```
func _on_share_btn_pressed():
	print(Engine.has_singleton("GodotShare"))
	print(Engine.get_singleton("GodotShare"))
```

And I get:

`False`
`[Object:null]`

I think this would happen with any Android module.

**Comment by TheTrope** [comment]
Definitly the same issue as mine #22932
Following it !

**Comment by wombatwingdings** [comment]
It's not just me then! Thanks all.

**Comment by s0ra** [comment]
I don't if this error related to this bug.

`AndroidRuntime: java.lang.ClassNotFoundException: Didn't find class "org/godotengine/godot/..." on path: DexPathList[[zip file "/system/framework/org.apache.http.legacy.boot.jar", zip file "/data/app/org.godotengine...`

version: godot-b05efd98ffe3036b05e501b26ed2ee811161ee11

**Comment by dumitru-stama** [comment]
Same for me, added PR #23021 to fix this

**Comment by akien-mga** [comment]
Please report this bug to https://github.com/FrogSquare/GodotAds, the issue is in their documentation. You should use `ProjectSettings`, not `Globals`, and that's since 3.0.

**Comment by DDru** [comment]
@akien-mga what did you mean by ProjectSettings? I tried to use Engine.get_singleton and added classes to the project file, but this did not change anything.

**Comment by wombatwingdings** [comment]
I don't use that module myself, but from my experience, you would need to add the following to your project.godot file (or add it in "Project Settings -> Android -> Modules" in the UI). The readme for your module doesn't mention this.
```
[android]
modules="org/godotengine/godot/GodotAds"
```

**Comment by DDru** [comment]
@wombatwingdings it does not work.

**Comment by burstina** [comment]
I'm using Godot 3.1
Engine.has_singleton() returns true
Engine.get_singleton() returns null

**Comment by Siler88** [comment]
hey guys. i still have the bug. issnt it fixed jet?

**Comment by ANeller** [comment]
Yeah im having the exact same problems and have also tried all the various tips above

**Comment by DDru** [comment]
@Siler88 @ANeller 
I found out that it works (returns not null) only on Android. Just run the game on your phone and look at the debug result.

**Comment by ANeller** [comment]
> @Siler88 @ANeller
> I found out that it works (returns not null) only on Android. Just run the game on your phone and look at the debug result.

I tried to run on my phone but doesnt load as soon as I add this to my engine config file - 

[android]
modules="org/godotengine/godot/GodotAds,org/godotengine/godot/AdMob,org/godotengine/godot/GDAdColony,org/godotengine/godot/GDChartboost,org/godotengine/godot/GDVungle,org/godotengine/godot/GDMopub,org/godotengine/godot/GDUnityAds"

is this correct? - https://screenpresso.com/=010vc

**Comment by DDru** [comment]
@ANeller 
Look at the manual here: https://github.com/FrogSquare/GodotAds
- First you need to add files to the source code and specify in config.py which Ad you need
- Then rebuild engine source code
- In the modules you specify only selected Ad

**Comment by ANeller** [comment]
@Dru thanks for the help, I tried all those steps and rebuilt the source code etc, but as soon as I add this part..

[android]
modules="org/godotengine/godot/GodotAds,org/godotengine/godot/AdMob,org/godotengine/godot/GDAdColony,org/godotengine/godot/GDChartboost,org/godotengine/godot/GDVungle,org/godotengine/godot/GDMopub,org/godotengine/godot/GDUnityAds"

to my engine.cfg (which I have been told is  my project.godot) file now, my project crashes on my phone (doesnt even load just goes black then closes)

Anyways I dont want to spam this thread up any longer, I'm not very good with this stuff

**Comment by ai-boson** [comment]
On the android device I get this bug report. I am using Godot 3.2.2.
![bugreportonphone](https://user-images.githubusercontent.com/72218574/103294706-6212c580-4a18-11eb-825f-a8e28b950105.PNG)
